state/modelgeneration.go

```diff
@@ -453,3 +436,7 @@ func newGeneration(st *State, doc *generationDoc) *Generation {
 		doc: *doc,
 	}
 }
+
+// ErrGenerationNoAutoComplete indicates that the generation can not be
+// automatically completed.
+var ErrGenerationNoAutoComplete = errors.New("generation can not be auto-completed")
```

Comment:
Perhaps this should reside in core/model/generation.go

Reply:
I think that depends on whether we want to allow ever having it bubble out of the local calls. Certainly if there is external code that is going to call AutoComplete via an interface{}, it would be better for them *not* to have to import 'state' just to get the possible error type.

The other possibility is that we just make AutoComplete() (bool, error) and the bool says whether autocomplete happened or not, and we don't treat it not happening as an Error condition.
As is, we only have a bool anyway (did we Cause()== NoAutoComplete is a boolean response, right?)